reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -379,13 +379,58 @@ class StaticHtmlReporter : Reporter() {
         h2 { +"Index" }
 
         ul {
-            reportTableModel.evaluatorIssues?.let {
-                li { a("#policy-violation-summary") { +"Rule Violation Summary (${it.size} violations)" } }
+            reportTableModel.evaluatorIssues?.let { issues ->
+                var errorCount = 0
+                var warningCount = 0
+                var hintCount = 0
+
+                issues.forEach {
+                    when (it.severity) {
+                        Severity.ERROR -> ++errorCount
+                        Severity.WARNING -> ++warningCount
+                        Severity.HINT -> ++hintCount
+                    }
+                }
+
+                li {
+                    a("#policy-violation-summary") {
+                        +"Rule Violation Summary ($errorCount errors, $warningCount warnings, $hintCount hints)"
+                    }
+                }
             }
 
-            val numberOfIssues = reportTableModel.issueSummary.rows.count()
-            if (numberOfIssues > 0) {
-                li { a("#issue-summary") { +"Issue Summary ($numberOfIssues issues)" } }
+            var errorCount = 0
+            var warningCount = 0
+            var hintCount = 0
+
+            reportTableModel.issueSummary.rows.forEach { issueRow ->
+                issueRow.analyzerIssues.forEach { (_, issues) ->
+                    issues.forEach {
+                        when (it.severity) {
+                            Severity.ERROR -> ++errorCount
+                            Severity.WARNING -> ++warningCount
+                            Severity.HINT -> ++hintCount
+                        }
+                    }
+                }
+
+                issueRow.scanIssues.forEach { (_, issues) ->
+                    issues.forEach {
+                        when (it.severity) {
+                            Severity.ERROR -> ++errorCount
+                            Severity.WARNING -> ++warningCount
+                            Severity.HINT -> ++hintCount
+                        }
+                    }
+                }
+            }
```

Comment:
Same here, e.g.:

```kotlin
val allIssues = reportTableModel.issueSummary.rows.flatMap {
    it.analyzerIssues.flatMap { (_, issues) -> issues } + it.scanIssues.flatMap { (_, issues) -> issues }
}
val errorCount = allIssues.count { it.severity == Severity.ERROR }
```

Adding these error counts to the `IssueRow` class would keep the presentation code here cleaner and make the data available to other table reports.

Reply:
I'll look into adding these numbers to `IssueRow`.